### DISCUSSION TITLE
PayPal Payment Gateway Package Ready for Testing for Medusa v2

### DISCUSSION THREAD

**Discussion by amaster507** [medusajs/medusa#9346] [RFC]
https://github.com/amaster507/medusa-payment-paypal

I know there are some errors with this package in the way it is currently built with the tsconfig, etc. The plan is for medusa to copy this repo and place it beside the stripe package that is in v2, so this repo mirrors that of stripe.

I do know there is at least one bug right now with events. See https://github.com/amaster507/medusa-payment-paypal/blob/1b345ea130412d42dfd9c614220d0febd272f236/src/core/paypal-base.ts#L352 and if you know how to get the event type to use in the switch please let me know.

```ts
        // FIXME: How to get the event from the webhook data?
        // const event = webhookData. ???
        const event = "unknown" as string

        switch (event) {
```

**Comment by Megrax** [comment]
Hi can I use this package to integrate PayPal with medusa 2.0 for now? Do at least the basic functions work?
